vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.h

```diff
@@ -36,12 +37,17 @@ class VectorData final : public Data {
   VectorData& operator=(const VectorData& vector_data);
   VectorData& operator=(VectorData&& vector_data) noexcept;
 
+  // Mathematical vector operations
   friend double operator*(const VectorData& lhs, const VectorData& rhs);
-
+  void VectorAddElementWise(const VectorData& v_add);
+  void VectorDivideByScalar(float scalar);
+  float VectorSumElements();
   void Normalize();
 
-  int GetDimensionCountForTesting() const;
+  int GetDimensionCount() const;
+  int GetNonZeroElementsCount();
 
+  const std::string GetVectorAsString() const;
   const std::vector<float>& GetValuesForTesting() const;
```

Comment:
Could we not change to GetStorageForTesting()? as then we don't nee GetDimensionCountForTesting, GetValuesForTesting etc.

Reply:
I can change this, but I don't fully understand the benefit. For the majority of use cases, only the dimension count is actually needed (in order to initialize a new VectorData when a new embedding is being created from text)